### PR TITLE
Increase depth of curvev2 factory pool, withdraw from bcrvIbBTC position for treasury ops

### PR DESCRIPTION
**Pull request by jalbrekt85**
**updated #34 to withdraw from bcrvIbBTC position from treasury ops and get wbtc to deposit into curve factory pool**

- running script will deposit 5k badger and equal amounts wbtc
- right now, the entire bcrvIbBTC position is withdraw to wbtc. should only required deposit amount from `calc_deposit_amount` be withdraw each go?
-copied curve class pr from #100 for zapper method

### CODE REVIEW

File: scripts/issue/34/issue34.py [hunk 1]
```diff
@@ -21,27 +21,43 @@ def calc_deposit_amount(amount):
...
+        registry.eth.sett_vaults.bcrvIbBTC, owner=safe.address
+    )
+    
+    crvIbBTC = safe.contract(registry.eth.treasury_tokens.crvIbBTC)
     
     safe.take_snapshot(tokens=[BADGER, WBTC])
     
+    # get amounts to deposit into factory pool
     badger_amount = AMOUNT_BADGER_ETH * 10**BADGER.decimals()
     wbtc_amount = calc_deposit_amount(badger_amount)
```
Comment (jorijnsmit) [inline scripts/issue/34/issue34.py:41]: get_dy cannot be used for finding out what the wbtc_amount is. the pool is currently not deep enough, and returns 23560388.

my suggestion here would be to do it manually and divide by price_oracle (and not price_scale). this way we add liquidity in a ratio that is truthful to the ratio we want to pool to be at.

**Review by jorijnsmit — CHANGES_REQUESTED** [review]
make sure to print a snapshot at the end and verify all numbers there make sense. this is what i am getting now:
```
snapshot result for 0x042B32Ac6b453485e357938bdC38e0340d4b9276:
                             balance_before              balance_after              balance_delta
symbol                                                                                           
BADGER            15,000.000000000000000000  10,000.000000000000000000  -5,000.000000000000000000
WBTC                   0.046680780000000000      25.363771770000000000      25.317090990000000000
bibbtc/sbtcCRV-f      25.000000000000000000       0.000000000000000000     -25.000000000000000000
ibbtc/sbtcCRV-f        0.508363299673756697       0.000000000000000000      -0.508363299673756697
BADGERWBTC-f           0.000000000000000000      38.553583438542914302      38.553583438542914302
```
whereas my expectations are roughly -1 bcrvibbtc and -5000 badger in return for ~77 pool tokens.

Screenshot 2022-02-10 at 23 08 47

**Review by jorijnsmit — DISMISSED** [review]
looks good now!
```
snapshot result for 0x042B32Ac6b453485e357938bdC38e0340d4b9276:
                             balance_before              balance_after              balance_delta
symbol                                                                                           
BADGER            15,000.000000000000000000  10,000.000000000000000000  -5,000.000000000000000000
WBTC                   0.046680780000000000       0.575674530000000000       0.528993750000000000
BADGERWBTC-f           0.000000000000000000      83.558976867363267849      83.558976867363267849
bibbtc/sbtcCRV-f      25.000000000000000000      23.602297466975601400      -1.397702533024398600
ibbtc/sbtcCRV-f        0.526529211261634753       0.000000000000000000      -0.526529211261634753 
```
~84 pool tokens in exchange for 5k badger and ~1.4 btc (5000/3700).

**Review by jorijnsmit — APPROVED** [review]

**Review by petrovska-petro — APPROVED** [review]
Neat usage of price_oracle to calc better ratio

As suggestion: 

Since, i believe we will do 3 chunks for increasing the depth, for the next iteration probably it is worthy to add more precise way to translate the value of bcrvIbBTC to wbtc (`wbtc_amount`), so we do not wd more than we should to keep it as dogfood on the meantime, as in this script there is left 0.52 idle

**Comment by jorijnsmit** [comment]
some weird merge conflict 🙄

**Review by petrovska-petro — APPROVED** [review]

**Pull request merged**